### ISSUE TITLE
Gradient explosion about harmful loss.

### ISSUE THREAD

**Issue by w-yibo** [domenicrosati/representation-noising#5]
Thanks for your great work.
I try the harmful loss in my code, but my experiment always failed because of gradient explosion. I also try the torch.log(1 + harmful loss), it still cause explosion. 
Is it necessary that use the layer-wise harmful loss? I just use the final output to compute harmful loss. Or do you have some suggestions?
